index.js

```diff
@@ -1,138 +1,155 @@
+  console.log('Processing ' + records.length + ' records')
 
   // retrieve token and schema
   Promise.all([token(), schema()])
     .then(function (res) {
-      var access_token = res[0];
-      var schema = res[1];
-      onReady(records, access_token, schema);
-    });
+      var accessToken = res[0]
+      var schema = res[1]
+      onReady(records, accessToken, schema)
+    })
 
   // run when access token and schema are loaded
-  function onReady(payload, access_token, schema) {
-    // console.log('Ready', access_token, schema);
+  function onReady (payload, accessToken, schema) {
+    console.log('Ready', accessToken, schema)
     // load avro schema
-    var avroType = avro.parse(schema);
+    var avroType = avro.parse(schema)
     // parse payload
-    var data = payload
-      .map(function(record){
-        return parseKinesis(record, avroType);
-      });
+    var records = payload
+      .map(function (record) {
+        return addSource(parseKinesis(record, avroType))
+      })
     // post to API
-    postRecords(access_token, data);
+    console.log('Posting records')
+    console.log(records)
+    postRecords(accessToken, records)
   }
 
   // map to records objects as needed
-  function parseKinesis(payload, avroType) {
+  function parseKinesis (payload, avroType) {
+    console.log('Parsing Kinesis')
     // decode base64
-    var buf = new Buffer(payload.kinesis.data, 'base64');
+    var buf = new Buffer(payload.kinesis.data, 'base64')
     // decode avro
-    var record = avroType.fromBuffer(buf);
-    return record;
+    var record = avroType.fromBuffer(buf)
+    return record
+  }
+
+  function addSource (record) {
+    console.log('Adding source')
+    record['nyplSource'] = 'sierra-nypl'
+    record['nyplType'] = process.env['NYPL_POST_TYPE']
+    return record
   }
 
   // bulk posts records
-  function postRecords(access_token, records) {
+  function postRecords (accessToken, records) {
     var options = {
       uri: process.env['NYPL_API_POST_URL'],
       method: 'POST',
-      headers: { Authorization: `Bearer ${access_token}` },
+      headers: { Authorization: `Bearer ${accessToken}` },
       body: records,
       json: true
-    };
+    }
 
     // POST request
-    request(options, function(error, response, body){
+    request(options, function (error, response, body) {
+      console.log('Posting...')
       if (error || body.errors && body.errors.length) {
         if (error) {
-          callback(new Error(error));
+          console.log('Error! ' + error)
+          callback(new Error(error))
         } else {
-          callback(new Error(body.errors[0]));
+          console.log('Error! ' + body.errors[0])
+          callback(new Error(body.errors[0]))
         }
-        // context.fail();
-
+        context.fail()
       } else {
-        callback(null, "POST Success");
-        // context.succeed();
+        callback(null, 'POST Success')
+        context.succeed()
```

Comment:
@ablwr @kfriedman I overlooked this. It's not needed for backward compatibility so it can be removed. The line before `callback(null, 'POST Success')` should stop the Lambda execution so `context.succeed()` is *probably* not called anyways.

@ablwr line 80 can be removed as well as line 77. The `fail()` state is called on lines 72 and 75.